docker/service/src/main/java/uk/nhs/adaptors/scr/services/AcsService.java

```diff
@@ -90,12 +90,22 @@ private Pair<String, String> getUserRoleCodeAndId(String authorisation, String n
 
         }
 
+        // if we haven't retrieved a role code from openID we try the SDS service
         try {
-            return Pair.of(sdsService.getUserRoleCode(nhsdSessionUrid), nhsdIdentity);
+            var roleCode = sdsService.getUserRoleCode(nhsdSessionUrid);
+            if (roleCode != null && !roleCode.isEmpty()) {
+                return Pair.of(roleCode, nhsdIdentity);
+            }
         } catch (BadRequestException | URISyntaxException e) {
+            LOGGER.info(String.format("Unable to determine Job Role Code for "
+                + "the given RoleID via the SDS Service: %s", nhsdSessionUrid));
             throw new BadRequestException(String.format("Unable to determine SDS Job Role Code for "
                 + "the given RoleID: %s", nhsdSessionUrid));
         }
+        LOGGER.info(String.format("Unable to determine Job Role Code for "
+            + "the given RoleID via the SDS Service: %s", nhsdSessionUrid));
+        throw new BadRequestException(String.format("Unable to determine SDS Job Role Code for "
+            + "the given RoleID: %s", nhsdSessionUrid));
```

Comment:
```suggestion
        String error = String.format("Unable to determine Job Role Code for the given RoleID via the SDS Service: %s", nhsdSessionUrid)
        LOGGER.info(error);
        throw new BadRequestException(error);
```

The two strings are slightly different but not meaningfully so I think. Same strings are used above too